### PR TITLE
fixed Mask2Former image processor segmentation maps handling

### LINKED ISSUE #33295: Mask2FormerImageProcessor - fails to process multichannel image
### System Info

- `transformers` version: 4.43.4
- Platform: Linux-6.5.0-1027-oem-x86_64-with-glibc2.35
- Python version: 3.11.9
- Huggingface_hub version: 0.24.6
- Safetensors version: 0.4.4
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.4.0 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: no
- Using GPU in script?: yes
- GPU type: NVIDIA RTX A4500 Laptop GPU

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

[Colab](https://colab.research.google.com/drive/1T_lGRfyRoQCycZbLJAVsa6c12K5NI7GD?usp=sharing)

### Expected behavior

The image processing should be handled correctly. 

It looks like the [get_max_height_width](https://github.com/huggingface/transformers/blob/174890280b340b89c5bfa092f6b4fb0e2dc2d7fc/src/transformers/models/mask2former/image_processing_mask2former.py#L938) is using a default parameter rather then a value `input_data_format` set in the class constructor.

### PR DESCRIPTION
**Pull request by maciej-adamiak**
Fixes segmentation maps padding size calculation when handling multichannel images using previously inferred `input_data_format`. 

Fixes #33295

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case -> #33295.
- [x] Did you write any new necessary tests?

## Who can review?
Anyone

### CODE REVIEW

**Comment by LysandreJik** [comment]
cc @amyeroberts @qubvel

File: tests/models/mask2former/test_image_processing_mask2former.py [hunk 1]
```diff
@@ -49,8 +51,8 @@ def __init__(
         size=None,
         do_resize=True,
         do_normalize=True,
-        image_mean=[0.5, 0.5, 0.5],
-        image_std=[0.5, 0.5, 0.5],
+        image_mean=0.5,
+        image_std=0.5,
```
Comment (qubvel) [inline tests/models/mask2former/test_image_processing_mask2former.py:55]: Please, revert to the original state :)

**Review by qubvel — COMMENTED** [review]
Thanks a lot for working on this and adding tests, just a nit comment

File: tests/models/mask2former/test_image_processing_mask2former.py (hunk 1 above)
Comment (maciej-adamiak) [reply tests/models/mask2former/test_image_processing_mask2former.py:55]: When a revert is applied both values will become 3-element lists and will not be compatible with tests run on images with a different number of channels.

Comment (maciej-adamiak) [reply tests/models/mask2former/test_image_processing_mask2former.py:55]: If we want to be super precise I can write '[0.5] * num_channels'.

Comment (qubvel) [reply tests/models/mask2former/test_image_processing_mask2former.py:55]: Yes, that way will be better, just to be sure normalization work for this case 👍

File: tests/models/mask2former/test_image_processing_mask2former.py [hunk 2]
```diff
@@ -242,6 +269,15 @@ def common(is_instance_map=False, segmentation_type=None):
...
         common(is_instance_map=True, segmentation_type="pil")
+        common(num_channels=1, numpify=True)
+        common(num_channels=2, numpify=True, input_data_format=ChannelDimension.LAST)
+        common(num_channels=5, numpify=True, input_data_format=ChannelDimension.LAST, do_resize=False)
+
+        with pytest.raises(ValueError, match="Unable to infer channel dimension format"):
+            common(num_channels=5, numpify=True, do_resize=False)
+
+        with pytest.raises(TypeError, match=r"Cannot handle this data type: .*"):
+            common(num_channels=5, numpify=True, input_data_format=ChannelDimension.LAST)
```
Comment (qubvel) [inline tests/models/mask2former/test_image_processing_mask2former.py:280]: Sorry, missed this for the first time, we use unittest for testing, can you please update to `with self.assertRaises(ValueError)` or `assertRaisesRegex` and remove `import pytest` on the top of the module.

File: tests/models/mask2former/test_image_processing_mask2former.py [hunk 3]
```diff
@@ -223,9 +234,25 @@ def test_with_size_divisor(self):
...
+            segmentation_type=None,
+            numpify=False,
+            num_channels=3,
+            input_data_format=None,
+            do_resize=True,
+        ):
+            self.image_processor_tester.num_channels = num_channels
+            self.image_processor_tester.do_resize = do_resize
+            self.image_processor_tester.image_mean = [0.5] * num_channels
+            self.image_processor_tester.image_std = [0.5] * num_channels
```
Comment (qubvel) [inline tests/models/mask2former/test_image_processing_mask2former.py:248]: not sure it's a good way to modify class properties in place, that may lead to some hard-to-debug issues later. You can recreate an instance here with an updated image processor dict.

File: tests/models/mask2former/test_image_processing_mask2former.py [hunk 4]
```diff
@@ -242,6 +269,15 @@ def common(is_instance_map=False, segmentation_type=None):
         common(is_instance_map=True)
         common(is_instance_map=False, segmentation_type="pil")
         common(is_instance_map=True, segmentation_type="pil")
+        common(num_channels=1, numpify=True)
+        common(num_channels=2, numpify=True, input_data_format=ChannelDimension.LAST)
+        common(num_channels=5, numpify=True, input_data_format=ChannelDimension.LAST, do_resize=False)
```
Comment (qubvel) [inline tests/models/mask2former/test_image_processing_mask2former.py:274]: Can you please make a call with explicit `input_data_format=ChannelDimension.FIRST` for better readability of the test?

**Review by qubvel — COMMENTED** [review]
Thanks for the update, added a few comments regarding tests

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing! With @qubvel's suggestions all looks good to me -- let's get his final approval and then we can merge :)

**Review by qubvel — APPROVED** [review]
Thanks for the update!
